sound/soc/sof/ipc4-topology.c

```diff
@@ -1028,6 +1028,48 @@ static int sof_ipc4_update_hw_params(struct snd_sof_dev *sdev, struct snd_pcm_hw
 	return 0;
 }
 
+static int sof_ipc4_get_output_format(struct snd_sof_dev *sdev,
```

Comment:
commit message:

````
The previous patch fixes the problem with setting the output format when
only one of the pipelines is active. But in the case of simultaneous
capture, if PCM 0 is started first to record 32-bit 2ch 48KHz audio, the
output format of the module-copier.N.1 will be set to the same.
````
This is difficult to follow, there is no "output format" for a copier in general, the output format is pin-specific. So with that, it's unclear why the problem is.

````
 When this is followed by 32-bit 2ch 48KHz audio recording on PCM 1, the efx2 module's
input format will need to be set to 16-bit. In order to allow this, always
invoke the ip_prepare op on all widgets even if they are already prepared.
````

And this part omits that it's really the module-copier.N.1 Pin1 that needs to be configured to output 16-bit PCM.